### PR TITLE
V22F-751 Error for not having chosen a valid black box now displays on a label instead of the apply button

### PR DESCRIPTION
**Pull request by VoidLeech**

### CODE REVIEW

**Review by suppergerrie2 — APPROVED** [review]
Colour could be changed to red to be more clear

**Comment by VoidLeech** [comment]
> Colour could be changed to red to be more clear

Pure red (0xFF0000) actually contrasts pretty poorly with the average color of that area (0x2D5068) (some of which is even red).
By contrast, an orange (0xFFAD34) works much better. Other colors that contrast well don't fit with a theme of something going wrong (white and a variety of light greens/blues/pinks)

**Review by suppergerrie2 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 1]
```diff
@@ -406,20 +407,20 @@ static void SelectBlackBox(void* /* dummy */)
...
     {
-        GfuiButtonSetText(s_scrHandle, m_blackBoxButton, MSG_BLACK_BOX_NOT_EXE);
+        GfuiLabelSetText(s_scrHandle, m_noBlackBoxLabel, MSG_NO_BLACK_BOX);
         return;
     }
     // Enforce that file ends in .exe
     if (std::strcmp(path.extension().string().c_str(), ".exe") != 0)
     {
-        GfuiButtonSetText(s_scrHandle, m_blackBoxButton, MSG_BLACK_BOX_NOT_EXE);
+        GfuiLabelSetText(s_scrHandle, m_noBlackBoxLabel, MSG_NO_BLACK_BOX);
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:416]: Why was the error message changed? I think the old message was better to reflect the error.

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp (hunk 1 above)
Comment (VoidLeech) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:416]: This was always the error that was displayed if you tried to apply without a black box selected.
The fact that the file must match *.exe is displayed in the message hint.

Comment (VoidLeech) [reply speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp]: I've made it more explicit that there is a hint available on that label, as well as making sure that the messages for not having selected a black box and just choosing a file that is ineligible differ.

File: speed-dreams/source-2.2.3/data/data/menu/ResearcherMenu.xml [hunk 2]
```diff
@@ -96,6 +96,20 @@
             <attstr name="pushed image" val="data/img/button-left-pushed.png"/>
         </section>
 
+        <!-- Although this is a label, this is not a static control as we want to change it-->
+        <section name="NoBlackBoxError">
+            <attstr name="type" val="label"/>
+            <attnum name="x" val="20"/>
+            <attnum name="y" val="205"/>
+            <!-- "(Hint) " + 37 Spaces so maxlength is set correctly-->
+            <attstr name="text" val="(Hint)                                      "/>
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/data/data/menu/ResearcherMenu.xml:105]: Having (Hint) in the label is unnecessary, should be empty normally and show an error if one occurs.
```suggestion
            <!-- "Label for showing error messages to the user, needs to be at least 50 characaters long -->
            <attstr name="text" val="                                                                                      "/>
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 3]
```diff
@@ -40,9 +41,9 @@
 
 // Messages for file selection
 #define MSG_BLACK_BOX_NORMAL_TEXT "Choose Black Box: "
-#define MSG_BLACK_BOX_NOT_EXE     "Choose Black Box: chosen file was not a .exe"
-#define MSG_APPLY_NORMAL_TEXT     "Apply"
-#define MSG_APPLY_NO_BLACK_BOX    "Apply | You need to select a valid Black Box"
+#define MSG_BLACK_BOX_NOT_EXE     "(Hint) You did not select a valid Black Box"
+#define MSG_NO_BLACK_BOX          "(Hint) You need to select a valid Black Box"
+#define MSG_ONLY_HINT             "(Hint)"
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:46]:
```suggestion
#define MSG_BLACK_BOX_NOT_EXE     "You did not select a valid Black Box"
#define MSG_NO_BLACK_BOX          "You need to select a valid Black Box"
#define MSG_ONLY_HINT             ""
```

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**